mail/api.py

```diff
@@ -114,9 +126,14 @@ def send_batch(cls, subject, body, recipients,  # pylint: disable=too-many-argum
             chunk_size (int): The maximum amount of emails to be sent at the same time
 
         Returns:
-            list: List of requests.Response HTTP response from Mailgun
+            list:
+                List of (recipients, requests.Response, exception) where
+                recipients is a list of recipient emails,
+                Response is an HTTP response from Mailgun, if there is one,
+                exception is an exception which occurred when attempting to mail, if there is one
```

Comment:
That handles the exception case but the caller will still have to do the work of checking the status codes of each response. The way the function is now, the user will need to check the statuses anyway. If they're using `send_batch` and not `send_individual_email` or one of the other related single email methods they should be prepared to deal with some percent of errors in the output anyway.